lib/OpenQA/Controller/Test.pm

```diff
@@ -87,26 +91,56 @@ sub list_ajax {
             scope => $scope,
             limit => 500,
         );
+        while (my $j = $jobs->next) { push(@ids, $j->id); }
+        $jobs->reset;
+        $result_stats = OpenQA::Schema::Result::JobModules::job_module_stats(\@ids);
+    }
+
+    my %settings;
+
+    my $query = $self->db->resultset("JobSettings")->search(
+        {
+            job_id => { in => \@ids },
+            key => { in => [qw/DISTRI VERSION ARCH FLAVOR BUILD MACHINE/] }
+        }
+    );
+    while (my $s = $query->next) {
+        $settings{$s->job_id . "-" . $s->key} = $s->value;
```

Comment:
Why to use flat hash with modified keys instead of i.e. hash of hashes ( $settings{$s->job_id}{'DISTRI'} etc.)
It took me some time to get why are you adding '-DISTRI' and others at :139